mk_script_tarfile

```diff
@@ -1,174 +1,257 @@
 #!/usr/bin/env python
 
-"""
-Usage:
-  ./mk_script_tarfile <number or dev>
+"""Usage:
+  ./mk_script_tarfile <versino string>
 
 Aim:
 
 Create the tar file for the contributed scripts package, using
-either the identifier "dev" or the number (the main CIAO version
-is picked up automatically from the ciao-x.y/ directory).
-
-For CIAO 4.9 one version is created, which has both Python 2.7 and
-3.5 directory names in it. This may turn out to be a bad idea,
-in which case two versions will be created. This is still the case
-for CIAO 4.10.
-
-Whilst we could automate the choice of number, it seems safer
-to make the user specify it manually.
+the dotted vresino string, which shuold be 4.14.0 or 4.14.dev
+(adjust versino appropriately).
 
 The VERSION file (ciao-<>/contrib/VERSION.CIAO_scripts) is not
 tracked by git.
 
+The approach taken by this script has changed - it was manual, then
+based on setuptools - but now with setuptools moving to a buildm
+rather than install, system we are going back to running things
+manually.
+
 Files or directories called tests are not included in the tar
 file. This may be changed in the future (it may be useful to
 be able to run tests from an installed version), but the tests
 are currently very ad-hoc so it's not worth it.
 
-The copy argument has been removed.
-
 """
 
-import sys
-import os
+from collections import defaultdict
 import glob
-import time
+import os
+from pathlib import Path
+import shutil
 import subprocess as sbp
+import sys
+import time
 
-def get_ciao_version():
-    """Set up the CIAO version values based on the ciao-x.y/
-    directory in the current working directory.
-    """
 
-    dnames = [dname for dname in glob.glob("ciao-*.*")
-              if os.path.isdir(dname)]
-    if len(dnames) == 0:
-        sys.stderr.write("ERROR: no directory matching ciao-*.* found\n")
-        sys.exit(1)
+def build_dist(version, dirname, outdir, python="python3.8"):
+    """Build contrib scripts package into a separate temp dir"""
 
-    elif len(dnames) != 1:
-        sys.stderr.write("ERROR: found {} ".format(len(dnames)) +
-                         "directories matching ciao-*.*!\n")
-        sys.exit(1)
+    outdir = Path(outdir)
+    outdir /= dirname
```

Comment:
I'm assuming that `/` is a python 3.9-ism?  This doesn't work with ciao-install 4.14 (py3.8)